generator/integration_tests/tests/golden_kitchen_sink_rest_metadata_decorator_test.cc

```diff
@@ -0,0 +1,307 @@
+#include "generator/integration_tests/tests/mock_golden_kitchen_sink_rest_stub.h"
+#include <gmock/gmock.h>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace golden_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::golden_internal::MockGoldenKitchenSinkRestStub;
+using ::testing::AnyOf;
+using ::testing::Contains;
+using ::testing::IsEmpty;
+
+Status TransientError() {
+  return Status(StatusCode::kUnavailable, "try-again");
+}
+
+TEST(KitchenSinkRestMetadataDecoratorTest, GenerateAccessToken) {
+  auto mock = std::make_shared<MockGoldenKitchenSinkRestStub>();
+  EXPECT_CALL(*mock, GenerateAccessToken)
+      .WillOnce([](rest_internal::RestContext& context,
+                   google::test::admin::database::v1::
+                       GenerateAccessTokenRequest const&) {
+        EXPECT_THAT(
+            context.GetHeader("x-goog-api-client"),
+            Contains(google::cloud::internal::ApiClientHeader("generator")));
+        EXPECT_THAT(context.GetHeader("x-goog-user-project"), IsEmpty());
+        EXPECT_THAT(context.GetHeader("x-goog-quota-user"), IsEmpty());
+        EXPECT_THAT(context.GetHeader("x-server-timeout"), IsEmpty());
+        EXPECT_THAT(context.GetHeader("x-goog-request-params"), IsEmpty());
+        return TransientError();
+      });
+
+  internal::OptionsSpan span(Options{});
+  GoldenKitchenSinkRestMetadata stub(mock);
+  rest_internal::RestContext context;
+  google::test::admin::database::v1::GenerateAccessTokenRequest request;
+  auto status = stub.GenerateAccessToken(context, request);
+  EXPECT_EQ(TransientError(), status.status());
+}
+
+TEST(KitchenSinkRestMetadataDecoratorTest, GenerateIdToken) {
+  auto mock = std::make_shared<MockGoldenKitchenSinkRestStub>();
+  EXPECT_CALL(*mock, GenerateIdToken)
+      .WillOnce(
+          [](rest_internal::RestContext& context,
+             google::test::admin::database::v1::GenerateIdTokenRequest const&) {
+            EXPECT_THAT(context.GetHeader("x-goog-api-client"),
+                        Contains(google::cloud::internal::ApiClientHeader(
+                            "generator")));
+            EXPECT_THAT(context.GetHeader("x-goog-user-project"),
+                        Contains("test-user-project"));
+            EXPECT_THAT(context.GetHeader("x-goog-quota-user"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-server-timeout"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-goog-request-params"), IsEmpty());
+            return TransientError();
+          });
+
+  internal::OptionsSpan span(
+      Options{}.set<UserProjectOption>("test-user-project"));
+  GoldenKitchenSinkRestMetadata stub(mock);
+  rest_internal::RestContext context;
+  google::test::admin::database::v1::GenerateIdTokenRequest request;
+  auto status = stub.GenerateIdToken(context, request);
+  EXPECT_EQ(TransientError(), status.status());
+}
+
+TEST(KitchenSinkRestMetadataDecoratorTest, WriteLogEntries) {
```

Comment:
This is really a `QuotaUserOptionTest`. Not so much a `WriteLogEntries` test.

I think it would be more useful to have one unit test per thing that the metadata generator does than one unit test per golden RPC.